src/psyclone/psyir/nodes/scoping_node.py

```diff
@@ -65,6 +65,7 @@ def __init__(self, children=None, parent=None, symbol_table=None):
             self._symbol_table = symbol_table
         else:
             self._symbol_table = self._symbol_table_class(self)
+        self._symbol_table._node = self
```

Comment:
Was this linking of table to ScopingNode simply missing before?

Reply:
It was done as a separate statement with a protected access after setting the symbol_table in containers and routines. And yes I found it was missing in a few places, and that lead to the lookups missing symbols as they couldn't recurse up.